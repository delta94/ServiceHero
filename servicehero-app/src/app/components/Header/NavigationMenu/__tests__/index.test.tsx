import React from 'react';
import { render } from '@testing-library/react';
import { Store } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import { configureAppStore } from 'store/configureStore';
import { NavigationMenu } from '../index';
import { BrowserRouter } from 'react-router-dom';

const renderComponent = (store: Store) =>
  render(
    <Provider store={store}>
      <BrowserRouter>
        <NavigationMenu />
      </BrowserRouter>
    </Provider>,
  );

describe('<NavigationMenu />', () => {
  let store: ReturnType<typeof configureAppStore>;

  beforeEach(() => {
    store = configureAppStore();
  });
  it('should match the snapshot', () => {
    const component = renderComponent(store);
    expect(component.container.firstChild).toMatchSnapshot();
  });
});
