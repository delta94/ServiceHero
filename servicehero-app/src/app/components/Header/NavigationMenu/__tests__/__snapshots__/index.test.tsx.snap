// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationMenu /> should match the snapshot 1`] = `
.c0 {
  background: transparent;
  border-bottom: 0px;
  font-size: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .ant-menu-item-selected a {
  color: #ffffff;
  opacity: 1;
}

<ul
  class="ant-menu ant-menu-dark c0 ant-menu-root ant-menu-horizontal"
  role="menu"
>
  <li
    class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
    role="menuitem"
    style="display: none;"
  >
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="ant-menu-submenu-title"
      role="button"
    >
      <span>
        ···
      </span>
      <i
        class="ant-menu-submenu-arrow"
      />
    </div>
  </li>
  <li
    class="ant-menu-item ant-menu-item-only-child ant-menu-item-selected"
    role="menuitem"
  >
    <a
      href="/"
    >
      Home
    </a>
  </li>
  <li
    class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
    role="menuitem"
    style="display: none;"
  >
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="ant-menu-submenu-title"
      role="button"
    >
      <span>
        ···
      </span>
      <i
        class="ant-menu-submenu-arrow"
      />
    </div>
  </li>
  <li
    class="ant-menu-item ant-menu-item-only-child"
    role="menuitem"
  >
    <a
      href="/profile"
    >
      Profile
    </a>
  </li>
  <li
    class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
    role="menuitem"
    style="visibility: hidden; position: absolute; display: none;"
  >
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="ant-menu-submenu-title"
      role="button"
    >
      <span>
        ···
      </span>
      <i
        class="ant-menu-submenu-arrow"
      />
    </div>
  </li>
</ul>
`;
